analysis/dataset_report.py

```diff
@@ -86,13 +89,42 @@ def is_boolean(series):
     return ((series == 0) | (series == 1)).all()
 
 
+def suppress(series, threshold):
+    """Replaces values in series less than threshold with missing values."""
+    series_copy = series.copy()
+    series_copy[series_copy < threshold] = numpy.nan  # in place
```

Comment:
If the threshold is 5, then we should make this <= threshold so that we are in line with [this](https://docs.opensafely.org/releasing-files/#redacting-counts-less-than-or-equal-to-5)

Reply:
I think this is worth unpicking. To restate: Current practice is that values `<=5` are suppressed; then the remaining values are rounded to the nearest `5`. As such:

* the presence of the missing value tells us that the raw value was `1`, `2`, `3`, `4`, or `5`;
* the presence of the value `5` tells us that the raw value was `6` or `7`;
* the presence of the value `10` tells us that the raw value was `8`, `9`, `10`, `11`, or `12`.

In order from easiest to hardest to guess, we have raw values of:

1. `6` or `7`
2. `1`, `2`, `3`, `4`, or `5`
3. `>=8`

So, current practice means that we have a group for which there is a .5 chance of successfully guessing the raw values; and, as you say, some confusion ("Why is there a `5` in my table?"). For the important group, of the smallest raw values, there is a .2 chance of successfully guessing the raw values.

If values `<5` were suppressed, then the raw value of `5` would move group. Again, in order from easiest to hardest to guess:

1. `5`, `6`, or `7`
2. `1`, `2`, `3`, or `4`
3. `>=8`

This means we have a group for which there is a .33 chance of successfully guessing the raw values; and no confusion. But for the important group, of the smallest raw values, there is a .25 chance of successfully guessing the raw values.

Where am I going with this? 🙂 I think that suppressing small numbers is required, and that `<=5` protects the important group more than `<5`, but rounding introduces confusion and is suggested ("consider rounding"). I'd like to make dataset-report useful to most people, most of the time, with low/no configuration. I could:

* use `<=5`. Matches required and suggested current practice, some confusion
* remove rounding.[^1] Matches required current practice, no confusion
* round _then_ suppress. Matches required and suggested current practice (but is, in effect, more stringent), no confusion

I'm leaning towards round _then_ suppress. What do you think, @LFISHER7?

[^1]: I'd add it again later, but it wouldn't be the default.

Reply:
See 11f02b9 for the change from `<` to `<=`. This ensures the reusable action is in step with the documentation.

Reply:
Thanks for breaking it down so clearly! Your interpretation of rounding with base 7 is correct, but given the above, I would also lean to rounding then suppressing, with the suppression threshold and rounding base both 5. This satisfies our disclosure rules whilst keeping the counts in factors that are familiar